cron/monthly/invoice-platform-fees.js

```diff
@@ -0,0 +1,231 @@
+      ORDER BY
+        t."createdAt"
+    )
+
+    SELECT
+      *
+    FROM
+      "platformFees"
+    UNION
+    SELECT
+      *
+    FROM
+      "platformTips";
+  `,
+    { replacements: { date: date.format('L') } },
+  );
+  const byHost = groupBy(pastMonthTransactions, 'HostCollectiveId');
+  const today = moment.utc();
+
+  for (const [hostId, hostTransactions] of entries(byHost)) {
+    const { HostName, currency } = hostTransactions[0];
+
+    let items = entries(groupBy(hostTransactions, 'source')).map(([source, transactions]) => ({
+      incurredAt: date,
+      amount: round(sumBy(transactions, 'amount')),
+      description: `${source}s`,
+    }));
+
+    const transactionIds = hostTransactions.map(t => t.id);
+    const totalAmount = sumBy(items, i => i.amount);
+    console.info(
+      `Host ${HostName} (#${hostId}) has ${hostTransactions.length} pending transactions and owes ${
+        totalAmount / 100
+      } (${currency})`,
+    );
+
+    // Credit the Host with platform tips collected during the month
+    await models.Transaction.create({
+      amount: totalAmount,
+      amountInHostCurrency: totalAmount,
+      CollectiveId: hostId,
+      // Pia's account
+      CreatedByUserId: 30,
+      currency: currency,
+      description: `Platform Fees and Tips collected in ${moment.utc().subtract(1, 'month').format('MMMM')}`,
+      FromCollectiveId: FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES.FromCollectiveId,
+      HostCollectiveId: hostId,
+      hostCurrency: currency,
+      netAmountInCollectiveCurrency: totalAmount,
+      type: TransactionTypes.CREDIT,
+    });
+
+    const host = await models.Collective.findByPk(hostId);
+    const connectedAccounts = await host.getConnectedAccounts({
+      where: { deletedAt: null },
+    });
+
+    let PayoutMethodId = FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES.PayoutMethodId;
+    if (connectedAccounts?.find?.(c => c.service === 'transferwise')) {
+      PayoutMethodId = 2955;
+    } else if (connectedAccounts?.find?.(c => c.service === 'paypal') && !host.settings?.disablePaypalPayouts) {
+      PayoutMethodId = 6087;
+    }
+
+    // Create the Expense
+    const expense = await models.Expense.create({
+      ...FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES,
+      PayoutMethodId,
+      amount: totalAmount,
+      CollectiveId: hostId,
+      currency: currency,
+      description: `Platform settlement for ${moment.utc().subtract(1, 'month').format('MMMM')}`,
+      incurredAt: today,
+      data: { isPlatformTipSettlement: true, transactionIds },
+      type: expenseTypes.INVOICE,
+      status: expenseStatus.APPROVED,
+    });
+
+    // Create Expense Items
+    items = items.map(i => ({
+      ...i,
+      ExpenseId: expense.id,
+      CreatedByUserId: FEES_ON_TOP_SETTLEMENT_EXPENSE_PROPERTIES.UserId,
+    }));
+    await models.ExpenseItem.bulkCreate(items);
+
+    // Attach CSV
+    const Body = json2csv(hostTransactions.map(t => omit(t, ['data'])));
```

Comment:
I found the CSV sample not ideal in the choice of columns featured. Can we consider something more compact?